ktor-server/ktor-server-test-host/jvm/src/io/ktor/server/testing/suites/CompressionTestSuite.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.testing.suites
+
+import io.ktor.application.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.features.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.server.engine.*
+import io.ktor.server.testing.*
+import io.ktor.utils.io.*
+import io.ktor.utils.io.jvm.javaio.*
+import java.io.*
+import java.util.zip.*
+import kotlin.test.*
+
+abstract class CompressionTestSuite<TEngine : ApplicationEngine, TConfiguration : ApplicationEngine.Configuration>(
+    hostFactory: ApplicationEngineFactory<TEngine, TConfiguration>
+) : EngineTestBase<TEngine, TConfiguration>(hostFactory) {
+
+    @Test
+    fun testLocalFileContentWithCompression() {
```

Comment:
Nit: I was thinking of how to make this similar-looking tests table-driven or at least extract some common code, but maybe it's not worth it as there are quite a few parameters to introduce.
Maybe let's leave it as it is.

Reply:
I'm experimenting with that on the client-side, and haven't found any good solutions yet.

Maybe we should look in different test frameworks, something like `spec`. But I'm not sure if they work with multiplatform.